src/main/java/com/amazon/ion/impl/macro/EncodingDirectiveReader.kt

```diff
@@ -0,0 +1,145 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.macro
+
+import com.amazon.ion.*
+import com.amazon.ion.impl.*
+import com.amazon.ion.impl.macro.MacroRef.Companion.byId
+
+/**
+ * Reads encoding directives from the given [IonReader].
```

Comment:
Can you add a comment about the potential reusability of this class? IIRC, there's also an encoding directive reader in `IonReaderContinuableCore` that is only compatible with that class. Is this one actually usable with more readers, and if so, which ones?

Reply:
I'll add a comment. There is one in `IonReaderContinuableCore` that is more complicated because it has to deal with continuability. I ultimately decided to pare this one down to only what's necessary for IonReader, rather than try to unify them. There isn't that much code here, and most of the complexity is in the MacroCompiler, which has been generalized for both readers.